draft-ietf-ppm-dap.md

```diff
@@ -2118,28 +2111,30 @@ Helper knows it can clean up its state.
 #### Recovering from Aggregation Step Skew {#aggregation-step-skew-recovery}
 
 `AggregationJobContinueReq` messages contain a `step` field, allowing
-Aggregators to ensure that their peer is on an expected step of DAP
-aggregation. In particular, the intent is to allow recovery from a scenario
-where the Helper successfully advances from step `n` to `n+1`, but its
-`AggregationJobResp` response to the Leader gets dropped due to something like
-a transient network failure. The Leader could then resend the request to have
-the Helper advance to step `n+1` and the Helper should be able to retransmit
-the `AggregationJobResp` that was previously dropped. To make that kind of
-recovery possible, Aggregator implementations SHOULD checkpoint the most recent
-step's prep state and messages to durable storage such that the Leader can
-re-construct continuation requests and the Helper can re-construct continuation
-responses as needed.
+Aggregators to ensure that their peer is on the expected step of DAP
+aggregation. In particular, the intent is to allow detection and recovery from a
+scenario where the Helper and the Leader have differing views of the current
+aggregation step, which may occur if an `AggregationJobResp` response to the
+Leader is dropped due to a transient network failure. The Leader could then
+resend the original `AggregationJobInitReq` message to restart the aggregation
+process, allowing aggregation to complete successfully. To make this kind of
+recovery possible, Leader implementations SHOULD retain enough information to
+reconstruct the `AggregationJobInitReq` message in the case that the aggregation
+process becomes desynchronized.
 
 When implementing an aggregation step skew recovery strategy, the Helper SHOULD
```

Comment:
Here we should repeat the line about "changing an aggregation job's parameters is illegal".

Reply:
OK. Perhaps more controversially, I changed the `SHOULD` to a `MUST` to match the `MUST` around "changing an aggregation job's parameters is illegal" -- I think these were effectively in conflict in terms of how required this was.

(Historically, I think we specified a `SHOULD` here to permit an implementation to ignore the request body on a re-send -- other than the `step` parameter, I suppose -- and blindly reply with the previous response. I'm OK dropping this because (a) no one has expressed interest in implementing this and (b) boy howdy, those semantics would be challenging to debug -- the response would potentially be completely independent of the request.)